scripts/cleanup-azure-resources.sh

```
@@ -650,6 +666,108 @@ delete_service_principals() {
     echo "  - Skipped (already deleted): ${skipped}"
 }
 
+# Find soft-deleted Key Vaults matching the prefix
+find_soft_deleted_vaults() {
+    local prefix="$1"
+
+    print_info "Searching for soft-deleted Key Vaults with prefix '${prefix}'..." >&2
+
+    local vaults_json
+    vaults_json=$(az keyvault list-deleted --query "[?starts_with(name, '${prefix}')].{name: name, location: properties.location, deletionDate: properties.deletionDate}" -o json 2>/dev/null)
+    if [[ $? -ne 0 ]]; then
```

Shellcheck (0.11.0)

[style] Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?.

(SC2181)